skrub/_datetime_encoder.py

```diff
@@ -323,3 +330,17 @@ def _check_params(self):
             raise ValueError(
                 f"'resolution' options are {allowed}, got {self.resolution!r}."
             )
+
+    if sklearn_below_1_6:
```

Comment:
@jeromedockes The issue was that the scikit-learn tests (that are skipped) issue an error at collection because `preserve_dtype` was not defined.

Somehow it was not raising in the past for some reason (but it could have raised if we run the test). So we don't need to inherit from `TransformerMixin`, just make a similar definition of the tags.